submissions/BlueLamantine/friends app/js/main.js

```diff
@@ -0,0 +1,352 @@
+          this.sortIndicators.name
+        );
+      },
+      [this.sortMethods.byAge]: () => {
+        this.uncheckRadioButton({buttonName : this.sortMethods.byName});
+        this.sortByContent(
+          selectedRadioButton.id,
+          this.sortByAgeTypes,
+          this.sortIndicators.age
+        );
+      },
+    };
+    return sortMethod[methodName]();
+  }
+
+  startFilter() {
+    this.usersByDefault = [...this.usersDataForFilter];
+
+    this.searchInput.addEventListener('input', ({ target }) => {
+      let userInput = target.value.toUpperCase();
+
+      const usersByEmail = this.usersDataForFilter.filter(user =>
+        user.email.toUpperCase().startsWith(userInput)
+      );
+      this.renderDataWithFilters.usersDataForRender = usersByEmail;
+      this.renderDataWithFilters.renderProfiles();
+    });
+
+    this.filtersPanel.addEventListener('change', ({ target }) => {
+      this.selectSortMethod(target);
+      this.renderDataWithFilters.usersDataForRender = this.usersDataForFilter;
+      this.renderDataWithFilters.renderProfiles();
+    });
+  }
+
+  sortByGender(gender) {
+    this.usersDataForFilter = [...this.usersByDefault];
+    const selectGender = {
+      allGenders: () => {
+        this.usersDataForFilter = this.sortedUsers = [...this.usersByDefault];
+      },
+      male: () => {
+        this.usersDataForFilter = this.usersDataForFilter.filter(
+          user => user.gender == this.sortByGenderTypes.male
+        );
+        this.sortedUsers = [...this.usersDataForFilter];
+      },
+      female: () => {
+        this.usersDataForFilter = this.usersDataForFilter.filter(
+          user => user.gender == this.sortByGenderTypes.female
+        );
+        this.sortedUsers = [...this.usersDataForFilter];
+      },
+    };
+
+    return selectGender[gender]();
+  }
+
+  sortByContent(sortType, sortTypes, sortBy) {
+    if (this.sortedUsers.length !== 0) {
+      this.usersDataForFilter = [...this.sortedUsers];
+    } else {
+      this.sortedUsers = [...this.usersByDefault];
+    }
+
+    const typesOfSort = {
+      [sortTypes.byDefault]: () => {
+        this.usersDataForFilter = [...this.sortedUsers];
+      },
+      [sortTypes.inAscending]: () => {
+        if (sortBy == this.sortIndicators.name)
+          this.usersDataForFilter.sort((a, b) =>
+            a.name.first.localeCompare(b.name.first)
+          );
+        if (sortBy == this.sortIndicators.age)
+          this.usersDataForFilter.sort((a, b) =>
+            a.dob.age.toString().localeCompare(b.dob.age.toString())
+          );
+      },
+      [sortTypes.inDescending]: () => {
+        if (sortBy == this.sortIndicators.name)
+          this.usersDataForFilter.sort((a, b) =>
+            b.name.first.localeCompare(a.name.first)
+          );
+        if (sortBy == this.sortIndicators.age)
+          this.usersDataForFilter.sort((a, b) =>
+            b.dob.age.toString().localeCompare(a.dob.age.toString())
+          );
```

Comment:
Sorting logic the same as at lines 289-296. Only the order of arguments is changed.

You can extract it into two functions and pass arguments as you need in them.

Reply:
Ok, fixed it!